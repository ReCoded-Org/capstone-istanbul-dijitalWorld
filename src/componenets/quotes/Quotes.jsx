import React from "react";
import { Container, Row, Col } from "react-bootstrap";

const Quotes = () => {
  return <div> </div>;
};

export default Quotes;
