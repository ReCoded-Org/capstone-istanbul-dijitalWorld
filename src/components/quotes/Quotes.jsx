import React from "react";
import { Col, Row, Container, Image } from "react-bootstrap";
import { useTranslation } from "react-i18next";
import safeImage from "../../images/safe_image.jpeg";
import leonImage from "../../images/leon-biss.jpg";
import kidsImage from "../../images/small-kids.png";
import "./Quotes.css";

const Quotes = () => {
  const { t } = useTranslation();
  return (
    <Container>
      <Row>
        <Col
          xl={6}
          lg={12}
          md={12}
          className="contentColumns"
          data-testid="column1"
        >
          <div className="contentContainer">
            <h2 id="sayHerName">
              <b>{t("home.quotes.sayHerName.part1")}</b>
              <br />
              {t("home.quotes.sayHerName.part2")}
              <br />
              <span>{t("home.quotes.sayHerName.part3")}</span>
            </h2>
          </div>
        </Col>
        <Col
          xl={6}
          lg={12}
          md={12}
          className="contentColumns"
          data-testid="column1"
        >
          <div className="contentContainer">
            <Image src={safeImage} fluid />
          </div>
        </Col>
      </Row>
      <Row>
        <Col
          xl={6}
          lg={12}
          md={12}
          className="contentColumns"
          data-testid="column2"
        >
          <div className="contentContainer">
            <Image src={leonImage} fluid />
          </div>
        </Col>
        <Col
          xl={6}
          lg={12}
          md={12}
          className="contentColumns"
          data-testid="column2"
        >
          <div className="contentContainer">
            <h2 id="openLetter">
              <u>
                {" "}
                {t("home.quotes.openLetter.part1")}
                <br />
                {t("home.quotes.openLetter.part2")}
              </u>
              <br />
              <span>{t("home.quotes.openLetter.part3")}</span>
            </h2>
          </div>
        </Col>
      </Row>
      <Row>
        <Col
          xl={6}
          lg={12}
          md={12}
          className="contentColumns"
          data-testid="column2"
        >
          <div className="contentContainer">
            <h2 id="fightForGirls">
              {t("home.quotes.girlsFight.part1")}
              <br />
              {t("home.quotes.girlsFight.part2")}
              <br />
              {t("home.quotes.girlsFight.part3")}
            </h2>
            <button type="button" data-testid="button">
              Donate Now
            </button>
          </div>
        </Col>
        <Col
          xl={6}
          lg={12}
          md={12}
          className="contentColumns"
          data-testid="column2"
        >
          <div className="contentContainer">
            <Image src={kidsImage} fluid />
          </div>
        </Col>
      </Row>
    </Container>
  );
};

export default Quotes;
