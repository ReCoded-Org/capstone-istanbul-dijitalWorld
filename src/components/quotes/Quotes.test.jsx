import React from "react";
import ReactDOM from "react-dom";
import { render } from "@testing-library/react";
import Quotes from "./Quotes";

import "@testing-library/jest-dom";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<Quotes />, div);
});

it("button renders correctly", () => {
  const { getByTestId } = render(<Quotes />);
  expect(getByTestId("button")).toHaveAttribute("type", "button");
});

it("column 1 have correct class name", () => {
  const { getByTestId } = render(<Quotes />);
  expect(getByTestId("column-1")).toHaveClass("content-columns");
});

it("column 2 have correct class name", () => {
  const { getByTestId } = render(<Quotes />);
  expect(getByTestId("column-2")).toHaveClass("content-columns");
});
