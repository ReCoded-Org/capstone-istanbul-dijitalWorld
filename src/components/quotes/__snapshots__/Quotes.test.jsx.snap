// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="container"
>
  <div
    className="row"
    data-testid="sayHerNameRow"
  >
    <div
      className="contentColumns col-xl-6 col-lg-12 col-md-12"
    >
      <h2
        className="sayHerName"
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        home.quotes.sayHerName.hashtag
      </h2>
      <h2
        className="sayHerName"
      >
        home.quotes.sayHerName.title
      </h2>
      <span
        className="sayHerNameAuthor"
      >
        home.quotes.sayHerName.author
      </span>
    </div>
    <div
      className="contentColumns col-xl-6 col-lg-12 col-md-12"
    >
      <img
        className="img-fluid"
        src="safe_image.jpeg"
        width="100%"
      />
    </div>
  </div>
  <div
    className="row"
    data-testid="anOpenLetterRow"
  >
    <div
      className="contentColumns col-xl-6 col-lg-12 col-md-12"
    >
      <img
        className="img-fluid"
        src="leon-biss.jpg"
      />
    </div>
    <div
      className="contentColumns col-xl-6 col-lg-12 col-md-12"
    >
      <div
        className="contentContainer"
      >
        <h2
          className="openLetter"
        >
          home.quotes.openLetter.title
        </h2>
        <span
          className="anonymous"
        >
          home.quotes.openLetter.author
        </span>
      </div>
    </div>
  </div>
  <div
    className="row"
    data-testid="fightForGirlsRow"
  >
    <div
      className="contentColumns col-xl-6 col-lg-12 col-md-12"
    >
      <h2
        className="fightForGirls"
      >
        home.quotes.girlsFight.title
      </h2>
      <button
        className="donateButton"
        data-testid="button"
        type="button"
      >
        home.quotes.girlsFight.button
      </button>
    </div>
    <div
      className="contentColumns col-xl-6 col-lg-12 col-md-12"
    >
      <img
        className="img-fluid"
        src="small-kids.png"
      />
    </div>
  </div>
</div>
`;
